src/main/java/com/cloudbees/plugins/credentials/impl/CertificateCredentialsImpl.java

```diff
@@ -369,17 +386,18 @@ public static class UploadedKeyStoreSource extends KeyStoreSource implements Ser
 
         /**
          * The old uploaded keystore.
+         * Still used for snapshot taking, with contents independent of Jenkins instance and JVM.
          */
         @CheckForNull
         @Deprecated
-        private transient Secret uploadedKeystore;
+        private Secret uploadedKeystore;
```

Comment:
create a different Implementation of the interface that is self contained for the snapshot.
having this here, will cause it to be persisted in addition to the secret bytes.